app/models/assignment.rb

```diff
@@ -70,4 +70,20 @@ def to_label
     label += " - #{period_start && I18n.l(period_start)}"
     label
   end
+
+  def hours_since_last_submitted
+    if submitted_at.present?
+      hours.where('created_at > ?', submitted_at)
+    else
+      hours
+    end
+  end
+
+  def feedbacks_since_last_submitted
+    if submitted_at.present? && volunteer.user
```

Comment:
Why do you check for the volunteer having a user here? If the volunteer has no user the query on `where(author: volunteer.user)` will simply be empty, because `volunteer.user` is `nil` in that case.